### ISSUE TITLE
What's the next level up in terms of RSE skills?

### ISSUE THREAD

**Issue by DamienIrving** [merely-useful/py-rse#623]
_TLDR: What is the answer to [this twitter thread](https://twitter.com/ZaneSelvans/status/1386408647480147969)?_

I was recently reading over [A National Agenda for Research Software](https://doi.org/10.5281/zenodo.4940274) and the authors essentially define three categories of research software authors (with my simple interpretation of what they are in brackets):

1. Researchers (i.e. writing analysis code for themselves)
2. Nascent research software engineers (writing research software shared/used among their team and/or a small user community)
3. Research software engineers (writing and maintaining critical and potentially complex research software with a large user base) 

I feel like the Software Carpentry lesson materials are aimed at category 1 and our book is aimed at category 2. So what about category 3? Are there a bunch of advanced topics we didn't include in our book that people need to learn to move from category 2 to 3? Are those advanced topics captured in a book that already exists? I guess what I'm asking is, do we expect that people can progress from 1 to 3 without doing a full software engineering degree and if so, are there resources to help people get from 2 to 3?

**Comment by gvwilson** [comment]
1. Most programmers in industry who get to 3 do so without the "benefit" of a software engineering degree: it's all on-the-job learn-by-imitation-and-googling.
2. I'd like to warm up https://drive.google.com/drive/folders/1kZ3JFO3GZwM7cmV1Syi7Pz0aQ5okfn6t some day; the talk I'm doing next week on software design for data scientists will hopefully turn into a new module for it. I'll retire some of the existing modules (e.g., workflow and automation) that are part of your "2"; what else should be in here to make it suitable for "3"?

**Comment by DamienIrving** [comment]
Thanks, @gvwilson.

1. I suspected as much.
2. It looks like most of the content of those slides is covered in our book... perhaps I'm underselling the book and it does in fact service "2" and "3"?!

**Comment by gvwilson** [comment]
I think we do a good job of getting 2 to 3, and I hope there will be some new stuff in the book even for (the very small percentage of) researchers who are already at level 3. I honestly don't know what to recommend after that, even to people in industry: https://aosabook.org/ was my attempt to provide more concentrated material, but it didn't move the needle noticeably. I've thought more than once about trying to organize an AOSA-like volume on "The Architecture of Open Science Applications", but so far my saner half has fended it off :-)
